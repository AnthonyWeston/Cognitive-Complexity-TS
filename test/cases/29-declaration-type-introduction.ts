function declaration<T extends true & false>();
function declaration<T>() {

}

abstract class ClassWithDeclarations {
    abstract abs<T extends true & false>(): void;
    abstract abs<T>(): void;

    method<T extends true & false>();
    method<T>() {

    }
}

class GenericClass<T extends true & false> {

}