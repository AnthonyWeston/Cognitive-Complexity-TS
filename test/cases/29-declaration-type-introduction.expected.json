{
    "29-declaration-type-introduction.ts": {
        "score": 4,
        "inner": [
            {
                "name": "declaration",
                "score": 1,
                "line": 1,
                "column": 1
            },
            {
                "name": "declaration",
                "score": 0,
                "line": 2,
                "column": 1
            },
            {
                "name": "ClassWithDeclarations",
                "score": 2,
                "line": 6,
                "column": 1,
                "inner": [
                    {
                        "name": "abs",
                        "score": 1,
                        "line": 7,
                        "column": 5
                    },
                    {
                        "name": "abs",
                        "score": 0,
                        "line": 8,
                        "column": 5
                    },
                    {
                        "name": "method",
                        "score": 1,
                        "line": 10,
                        "column": 5
                    },
                    {
                        "name": "method",
                        "score": 0,
                        "line": 11,
                        "column": 5
                    }
                ]
            },
            {
                "name": "GenericClass",
                "score": 1,
                "line": 16,
                "column": 1
            }
        ]
    }
}